### PR TITLE
[Bug] [Core] Fix unintentionally pressed modifiers when key up on using key override

### PR DESCRIPTION
**Pull request by kjuq**
## Description

In the status quo, the trigger_mods is repressed before the replacement_key is up like the Picture 1 (SQ). This behavior causes some problems for example overridden arrow keys are unable to navigate search suggestions on YouTube.
My PR solves this issue as Picture 2 (AP).

### Picture 1 (SQ)

Picture 1 (SQ)

### Picture 2 (AP)

Picture 2 (AP)

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* nothing

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by sigprof** [comment]
> In the status quo, the trigger_mods is repressed before the replacement_key is up like the Picture 1 (SQ).

Actually it seems that the modifier changes are accumulated and sent in the same report as the release of the overriding key, but the host processes the modifier change before the regular key release.  (In general, the key override code seems to be trying hard to decrease the number of sent reports by combining the modifier changes with normal key changes, which might not be actually good in some cases.)

File: quantum/process_keycode/process_key_override.c [hunk 1]
```diff
@@ -190,11 +184,17 @@ const key_override_t *clear_active_override(const bool allow_reregister) {
...
             unregister_code(mod_free_replacement);
         }
+        send_keyboard_report();
     }
 
+    // Clear the suppressed mods
+    clear_suppressed_override_mods();
+
+    // Unregister the replacement. First remove the weak override mods
+    clear_weak_override_mods();
```
Comment (sigprof) [inline quantum/process_keycode/process_key_override.c:196]: This comment is no longer correct after moving the code.

Comment (kjuq) [reply quantum/process_keycode/process_key_override.c]: Thank you for your review. I've fixed the comment.

**Comment by kjuq** [comment]
> (In general, the key override code seems to be trying hard to decrease the number of sent reports by combining the modifier changes with normal key changes, which might not be actually good in some cases.)

Is it so important to decrease the number of the reports, maybe is it helpful to improve performance or saving memory? Actually, I'm very newbie for programming in this kind of resource-limited computer, so I'm just curious.

**Comment by drashna** [comment]
> > (In general, the key override code seems to be trying hard to decrease the number of sent reports by combining the modifier changes with normal key changes, which might not be actually good in some cases.)
> 
> Is it so important to decrease the number of the reports, maybe is it helpful to improve performance or saving memory? Actually, I'm very newbie for programming in this kind of resource-limited computer, so I'm just curious.

While sending the keyboard report is one of the more expensive things the firmware does (IIRC), I think that some of the concern here is that sending the mod changes with normal key changes is one of the things that leads to issues on some systems.  For instance, the RDP issues may be caused by the quick succession, or even being sent at the same time.  This could compound that issue. 


Also, it would be great to see some unit tests added, as it  would help instill confidence in the code changes. There are a number of simpler features that have unit tests that could be used for reference (caps words, tri layer, secure, etc).

**Comment by kjuq** [comment]
Okay. I will try to write unit test.

**Pull request closed without merging**